xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass
+import subprocess
+import re
+import os
+from typing import Dict, List
+
+# Path to compiled C++ executable
+xml_converter: str = "../build/xml_converter"
+
+
+@dataclass
+class Attribute_Tests:
+    attribute_name: str
+    attribute_path: str
+    tests: List[str]
+    output_result: Dict[str, List[str]]
+    expect_output_to_be_equal: Dict[str, bool]
+    expect_files_to_be_equal: Dict[str, bool]
+
+
+def run_xml_converter(*args):
+    try:
+        # Build the command to execute the C++ program with the desired function and arguments
+        cmd = [xml_converter] + list(args)
+
+        # Run the C++ program and capture its output
+        result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        # Check if the C++ program executed successfully
+        if result.returncode == 0:
+            return result.stdout
+        else:
+            return result.stderr
+    except Exception as e:
+        print("Error:", e)
+        return None
+
+
+def are_files_equal(file_path1, file_path2) -> bool:
+    try:
+        with open(file_path1, 'rb') as file1, open(file_path2, 'rb') as file2:
+            content1 = file1.read()
+            content2 = file2.read()
+
+        return content1 == content2
+
+    except (FileNotFoundError, PermissionError):
+        print(f"Error opening file {file_path1} or {file_path2}")
```

Comment:
you can to `as e` at the end of the except and then print the real error too.

Reply:
Good idea. Done